cmake/Corrosion.cmake

```diff
@@ -1086,8 +1085,29 @@ function(corrosion_link_libraries target_name)
             $<TARGET_PROPERTY:${library},LINKER_LANGUAGE>
         )
 
-        corrosion_add_target_local_rustflags(${target_name} "-L$<TARGET_LINKER_FILE_DIR:${library}>")
-        corrosion_add_target_local_rustflags(${target_name} "-l$<TARGET_LINKER_FILE_BASE_NAME:${library}>")
+        if (TARGET ${library})
+            corrosion_add_target_local_rustflags(${target_name}
+                "-L$<TARGET_LINKER_FILE_DIR:${library}>"
+                "-l$<TARGET_LINKER_FILE_BASE_NAME:${library}>"
+            )
+            add_dependencies(_cargo-build_${target_name} ${library})
+            get_property(libs TARGET ${library} PROPERTY INTERFACE_LINK_LIBRARIES)
+            corrosion_link_libraries(${target_name} ${libs})
+        elseif(IS_ABSOLUTE ${library})
+            # Linking via full path
+            corrosion_add_target_local_rustflags(${target_name} "-Clink-arg=${library}")
+        else()
+            # We have to assume ${library} is a non-CMake library name
+            corrosion_add_target_local_rustflags(${target_name} "-l${library}")
+        endif()
+
+        # Propagate dependency on ${library}
+        set_property(
+            TARGET _cargo-build_${target_name}
+            APPEND
+            PROPERTY INTERFACE_LINK_LIBRARIES
```

Comment:
`_cargo-build_${target_name}` is a custom target, so I don't believe this property will have any effect. 
Is your intention here to add a rule to trigger rebuilds, or to propogate the dependencies to C/C++ consumers of a Rust static library ? If it's the latter, then https://github.com/corrosion-rs/corrosion/pull/506 should be sufficient.